Sources/MapboxCoreNavigation/PassiveLocationManager.swift

```diff
@@ -97,18 +97,22 @@ open class PassiveLocationManager: NSObject {
     public weak var delegate: PassiveLocationManagerDelegate?
     
     /**
-     Starts the generation of location updates. 
+     Starts the generation of location updates and resumes trip session.
+
+     To learn more read `PassiveLocationManager.resumeTripSession()` method documentation.
      */
+    @available(*, deprecated, message: "Use resumeTripSession() instead.")
```

Comment:
Why not remove this method completely? There will be breaking changes anyway. Also, please update `PassiveLocationManager.init` code doc postcondition as it recommends this method.

Less related question: while we are cleaning up the API, why not leave this method name but replace it's contents with `resumeTripSession`? This might be less declarative that this action is related to billing conditions, but then we could avoid introducing pair of new methods and won't end up with `startUpdatingLocation/resumeTripSession/pauseTripSession` but will have just `start/pauseUpdatingLocation`

Reply:
@Udumft Thanks for these questions:

> Why not remove this method completely

Because of no breaking change policy mostly. 

> There will be breaking changes anyway.

Yeah, it seems that RC still can contain breaking changes, although I've been told otherwise. 

> Also, please update PassiveLocationManager.init code doc postcondition as it recommends this method.

Thanks

> Less related question: While cleaning up the API, why not leave this method name but replace its contents with resumeTripSession? It might be less declarative that this action is related to billing conditions, but then we could avoid introducing pair of new methods and won't end up with startUpdatingLocation/resumeTripSession/pauseTripSession but will have start/pauseUpdatingLocation

From my point of view, it won't be evident that pausing location updates have more side effects. I expect us to have more sessions like API in the future, like pausing navigator, sessions for telemetry, billing sessions won't go anywhere, and we want to provide a means of controlling it. So my idea is that there will be a generic name that won't embed the implementation detail, which is a good thing for library evolution. 

@mapbox/navigation-ios We can debate the rename of "pause/resume trip session" to something else. For example: "start/stop".

Reply:
I renamed pause/resume trip session to `startTripSession` and `stopTripSession`. Along with this, trip session doesn't start with `PassiveLocationManager` instantiation (this was weird behaviour in the first place). This is a safe change because:
- `startUpdatingLocation` (`startTripSession` now) was required to be called after creating `PassiveLocationManager` and Maps calls this method through `PassiveLocationProvider`. 
- We mention the requirement to call this method in `init` `postcondition`. 

I think this simplifies this class usage and adoption.

Also removed startUpdatingLocation instead of deprecating it.